CMakeLists.txt

```diff
@@ -1,12 +1,13 @@
 cmake_minimum_required(VERSION 3.27)
 project(safe LANGUAGES CXX)
 
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-add_custom_target(copy_compile_commands ALL
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different
-    ${CMAKE_BINARY_DIR}/compile_commands.json
-    ${CMAKE_SOURCE_DIR}/compile_commands.json
-    DEPENDS ${CMAKE_BINARY_DIR}/compile_commands.json)
+# TODO: Maybe get rid of? Don't remember why its here and I think libhal's cmake utils handles this
+# set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+# add_custom_target(copy_compile_commands ALL
+#     COMMAND ${CMAKE_COMMAND} -E copy_if_different
+#     ${CMAKE_BINARY_DIR}/compile_commands.json
+#     ${CMAKE_SOURCE_DIR}/compile_commands.json
+#     DEPENDS ${CMAKE_BINARY_DIR}/compile_commands.json)
```

Comment:
This copies the compile_command.json file to the root of the directory. This is useful because clangd can only look at one thing at a time and each build type (release, debug) has a different build directory. In this way, we always overwrite the previous compile_commands.json with the latest build that was executed. Everything should work about the same regardless of the build type unless you are testing something between host software and firmware where the registers sizes and object sizes change.

I'd recommend bringing this back so you have clangd support automatically with each build.